finagle-netty4/src/main/scala/com/twitter/finagle/netty4/ssl/server/Netty4ServerSslHandler.scala

```diff
@@ -8,9 +13,74 @@ import com.twitter.finagle.ssl.server.{
   SslServerSessionVerifier
 }
 import com.twitter.finagle.transport.Transport
-import com.twitter.finagle.Stack
-import io.netty.channel.{Channel, ChannelInitializer, ChannelPipeline}
-import io.netty.handler.ssl.SslHandler
+import com.twitter.util.{Future => TwitterFuture}
+import io.netty.buffer.ByteBufAllocator
+import io.netty.channel._
+import io.netty.handler.ssl._
+
+final case class SniSupport(mapping: SniSupport.ServerNameToContext)
```

Comment:
For `mapping`, I believe we should use `Option[Map[String, SslServerConfiguration]]` instead.
None means the parameter is off.
Some(sniMapping) is a mapping from hostnames to SslServerConfigurations